frontend/src/js/components/login/signup-steps/UserdataEntry.tsx

```diff
@@ -21,35 +21,50 @@ import TextInput from '@northern.tech/common-ui/forms/TextInput';
 
 import { OAuthHeader } from '../Login';
 
-export const UserDataEntry = ({ classes, onProgessClick }) => {
-  const { isValid } = useFormState();
-  const email = useWatch({ name: 'email' });
+const defaultValues = { email: '', password_confirmation: '', password: '' };
 
-  const handleKeyPress = ({ key }) => {
-    if (key === 'Enter') {
-      onProgessClick();
+export type UserData = {
+  email: string;
+  password: string;
+  password_confirmation: string;
+};
+
+export const UserDataEntry = ({ classes, onSubmit }) => {
+  const methods = useForm({ mode: 'onSubmit', defaultValues });
+  const { handleSubmit, trigger, watch } = methods;
+  const email = watch('email');
+  const password = watch('password');
+  const passwordConfirmation = watch('password_confirmation');
+  const isNotDefined = !(email && password && passwordConfirmation);
+
+  const onFormBlur = ({ target: { id } }) => {
+    if (id !== 'password_confirmation') {
+      return;
     }
+    return trigger();
   };
 
   return (
-    <div className={classes.userData} onKeyPress={handleKeyPress}>
-      <h1 className="flexbox centered">Create your account</h1>
-      <OAuthHeader type="Sign up" />
-      <TextInput hint="Email *" label="Email *" id="email" required={true} validations="isLength:1,isEmail,trim" />
-      <PasswordInput
-        id="password"
-        label="Password *"
-        validations={`isLength:8,isNot:${email}`}
-        create={true}
-        generate={false}
-        required={true}
-        className="margin-bottom-small"
-      />
-      <PasswordInput id="password_confirmation" label="Confirm password *" validations={`isLength:8,isNot:${email}`} required={true} />
-      <Button className="margin-top" color="primary" disabled={!isValid} variant="contained" onClick={onProgessClick}>
-        Sign up
-      </Button>
-    </div>
+    <FormProvider {...methods}>
+      <form className={classes.userData} id="signup-form" noValidate onBlur={onFormBlur} onSubmit={handleSubmit(onSubmit)}>
```

Comment:
do we need `id="signup-form"` for this form?

Reply:
right, I guess I still had our `<Form />` in mind